### ISSUE TITLE
Bug: [no-unnecessary-type-conversion] rule reports on enum types which should be treated as opaque

### ISSUE THREAD

**Issue by davidhouweling** [typescript-eslint/typescript-eslint#11472]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=LAKAxg9gdgzgLgAjAV3hAtgSQCYILwIDkAltgIyEDcooAplMuggMKpwY4wIDeoCCOMviKkKAGj4DsAJmEkZhCSAC%2BNcNHhIMABwCGAJ1oA5CABUIAZTj7iUAObCUaLLjxuWbDthgA6QdXVYREh0PUMAdWI4AAtzKxt7R08XfHdWZ04-ch92eNs7AAoASgCgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0ipWmQ5MAbomjJKM9FAXQO0SABpw2PBPJUa9ZIxbsuPYQDNE9JrAC2Ujk%2BIjKyFRkjrNkHoAviBBQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const customId = 'id1';

enum CustomIds {
  Id1 = 'id1',
  Id2 = 'id2',
}

const compareNoToString = customId === CustomIds.Id1;
const compareWithToString = customId === CustomIds.Id1.toString();
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-unnecessary-type-conversion": "error",
    "@typescript-eslint/no-unsafe-enum-comparison": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No collision with rules...

### Actual Result

Line 9 shows what happens when you do not include `.toString()` resulting in `@typescript-eslint/no-unsafe-enum-comparison` throwing...
Line 10 shows what happens when you do add `.toString()` to resolve the issue on line 9... resulting in `@typescript-eslint/no-unnecessary-type-conversion` throwing, which it wants the code from line 9.

### Additional Info

Last week https://github.com/typescript-eslint/typescript-eslint/issues/11374 added `@typescript-eslint/no-unnecessary-type-conversion`  to strict checks, which has resulted in a failure to build (dependabot) as I have `@typescript-eslint/no-unsafe-enum-comparison` set to `error` in my eslint config already.

**Comment by bradzacher** [comment]
`no-unnecessary-type-conversion` probably shouldn't report on enum types given they're supposed to be opaque.

**Comment by skondrashov** [comment]
`no-unnecessary-template-expression` behaves the same way, which I think is also an issue.

**Comment by thebolarin** [comment]
@bradzacher This looks like a good issue for a first contribution. I would like to take it on if you don't mind
